go-controller/pkg/allocator/pod/pod_annotation.go

```diff
@@ -244,10 +252,16 @@ func allocatePodAnnotationWithTunnelID(
 	)
 
 	// Reconcile IPAM claim
-	if ipamClaim != nil && err == nil {
-		newIPAMClaim := ipamClaim.DeepCopy()
-		newIPAMClaim.Status.IPs = util.StringSlice(podAnnotation.IPs)
-		err = claimsReconciler.Reconcile(ipamClaim, newIPAMClaim, ipAllocator)
+	if ipamClaim != nil {
+		updatedIPAMClaim := updateIPAMClaimStatus(ipamClaim, podAnnotation, pod.Name, err)
+		allocationErr := err
+		reconcileErr := claimsReconciler.Reconcile(ipamClaim, updatedIPAMClaim, ipAllocator)
+		// Reconcile error takes precedence
+		if reconcileErr != nil {
```

Comment:
perhaps we only want to log in case of error? would appreciate opinions here..

Reply:
do we want to crash loop the pod ? if not, only log. 

Is this reconcilable ? I think so - i.e. the workload that previously had the "reserved" IP was removed.